packages/sync/src/Replicastore.php

```diff
@@ -1288,10 +1288,15 @@ public function checksum_histogram( $object_type, $buckets, $start_id = null, $e
 			$where .= " AND $id_field <= " . intval( $end_id );
 		}
 
+		$distinct = '';
+		if ( 'term_relationships' === $object_type ) {
+			$distinct = 'DISTINCT';
+		}
+
 		do {
 			list( $first_id, $last_id ) = $wpdb->get_row(
 				// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-				"SELECT MIN($id_field) as min_id, MAX($id_field) as max_id FROM ( SELECT $id_field FROM $object_table WHERE $where AND $id_field > $previous_max_id ORDER BY $id_field ASC LIMIT $bucket_size ) as ids",
+				"SELECT MIN($id_field) as min_id, MAX($id_field) as max_id FROM ( SELECT $distinct $id_field FROM $object_table WHERE $where AND $id_field > $previous_max_id ORDER BY $id_field ASC LIMIT $bucket_size ) as ids",
```

Comment:
AFAIK we're chunking here for performance and memory limitation reasons. But then, when expanding, we might end up with some objects that have just 1 term relationship, but others with hundreds. This might result in us hitting memory limits when we expand object IDs to multiple term relationships, because we don't know how much relationships each of the objects has. How would we prevent from that?

Reply:
I don't know.